File: DecompiledGameScripts/cyberpunk/devices/homeAppliances/toilet/toilet.swift

public class Toilet extends InteractiveDevice {

  protected cb func OnRequestComponents(ri: EntityRequestComponentsInterface) -> Bool {
    super.OnRequestComponents(ri);
  }

  protected cb func OnTakeControl(ri: EntityResolveComponentsInterface) -> Bool {
    super.OnTakeControl(ri);
    this.m_controller = EntityResolveComponentsInterface.GetComponent(ri, n"controller") as ToiletController;
  }

  protected cb func OnFlush(evt: ref<Flush>) -> Bool {
    if evt.IsStarted() {
      GameObject.PlaySoundEvent(this, this.GetDevicePS().GetFlushSFX());
      GameObjectEffectHelper.StartEffectEvent(this, this.GetDevicePS().GetFlushVFX());
    } else {
      GameObjectEffectHelper.StopEffectEvent(this, this.GetDevicePS().GetFlushVFX());
    };
    this.UpdateDeviceState();
  }

  public const func GetDevicePS() -> ref<ToiletControllerPS> {
    return this.GetController().GetPS();
  }

  protected const func GetController() -> ref<ToiletController> {
    return this.m_controller as ToiletController;
  }

  public const func DeterminGameplayRole() -> EGameplayRole {
    return EGameplayRole.GenericRole;
  }
}

File: Definitions/Game_Definitions/classes/Toilet.lua
---@meta
---@diagnostic disable

---@class Toilet : InteractiveDevice
Toilet = {}

---@return Toilet
function Toilet.new() return end

---@param props table
---@return Toilet
function Toilet.new(props) return end

---@param evt Flush
---@return Bool
function Toilet:OnFlush(evt) return end

---@param ri entEntityRequestComponentsInterface
---@return Bool
function Toilet:OnRequestComponents(ri) return end

---@param ri entEntityResolveComponentsInterface
---@return Bool
function Toilet:OnTakeControl(ri) return end

---@return EGameplayRole
function Toilet:DeterminGameplayRole() return end

---@return ToiletController
function Toilet:GetController() return end

---@return ToiletControllerPS
function Toilet:GetDevicePS() return end

